python/networkx.cpp

```diff
@@ -305,11 +305,23 @@ void add_key_value( //
     std::string_view column_name) {
     auto numeric_array = std::static_pointer_cast<array_type_at>(array);
 
-    std::string json_str;
+    std::string jsons_str;
+    jsons_str.reserve(es.size() * (column_name.size() + 2));
+    std::vector<ustore_length_t> offsets(es.size() + 1);
+
     for (size_t idx = 0; idx != es.size(); idx++) {
-        json_str = fmt::format("{{\"{}\": {}}}", column_name, numeric_array->Value(idx));
-        g.relations_attrs[es[idx].id].assign(value_view_t(json_str)).throw_unhandled();
+        offsets[idx] = jsons_str.size();
+        jsons_str += fmt::format("{{\"{}\": {}}}", column_name.data(), numeric_array->Value(idx));
```

Comment:
This can be expensive — format to the pre-reserved string instead of allocating new ones.